### PR TITLE
Configure CI for PRs

### PR DESCRIPTION
**Pull request by MaksimZhukov**
- Configure CI to automatically run builds on PRs
- Change yml files name from `xxx_xxx.yml` to `xxx-xxx.yml`
- Remove `Publish_GitHub_Release` stage

### CODE REVIEW

File: azure-pipelines/run-builds-job.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+trigger: none
+pr:
+  autoCancel: false
```
Comment (maxim-lobanov) [inline azure-pipelines/run-builds-job.yml]: Why autoCancel is false?

File: azure-pipelines/run-builds-job.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  branches:
+    include:
+    - master
+
+jobs:
+- job: Run_Builds
+  pool:
+    name: Azure Pipelines
+    vmImage: 'vs2017-win2016'
```
Comment (maxim-lobanov) [inline azure-pipelines/run-builds-job.yml]: Let's use `ubuntu-latest` because it contains more VMs

File: azure-pipelines/run-builds-job.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  steps:
+  - checkout: self
+    submodules: true
+
+  - task: PowerShell@2
+    displayName: 'Run build'
+    inputs:
+      targetType: filePath
+      filePath: './helpers/run-builds.ps1 '
+      arguments: '-AzDoOrganizationName $(ORGANIZATION_NAME) `
```
Comment (maxim-lobanov) [inline azure-pipelines/run-builds-job.yml]: These variables are available via in build definition. No need to hardcode them/

File: azure-pipelines/run-builds-job.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+                   -AzDoProjectName $(PROJECT_NAME) `
+                   -AzDoAccessToken $(ACCESS_TOKEN) `
```
Comment (maxim-lobanov) [inline azure-pipelines/run-builds-job.yml]: Is it possible to use System.AccessToken?

File: helpers/run-builds.ps1 [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    [Parameter(Mandatory)] [string] $AzDoProjectName,
+    [Parameter(Mandatory)] [string] $AzDoAccessToken,
+    [Parameter(Mandatory)] [string] $SourceBranch,
+    [Parameter(Mandatory)] [string] $PythonVersions,
+    [Parameter(Mandatory)] [UInt32] $BuildId
+)
+
+function Get-RequestParams {
+    param (
+        [Parameter(Mandatory)] [string] $PythonVersion
```
Comment (maxim-lobanov) [inline helpers/run-builds.ps1]: Let's do this script a bit more generic and avoid using `Python` word

File: azure-pipelines/run-builds-job.yml (hunk 1 above)
Comment (MaksimZhukov) [reply azure-pipelines/run-builds-job.yml]: Thanks!

File: azure-pipelines/run-builds-job.yml (hunk 2 above)
Comment (MaksimZhukov) [reply azure-pipelines/run-builds-job.yml]: Thanks!

File: azure-pipelines/run-builds-job.yml (hunk 3 above)
Comment (MaksimZhukov) [reply azure-pipelines/run-builds-job.yml]: Thanks!

File: helpers/run-builds.ps1 (hunk 5 above)
Comment (MaksimZhukov) [reply helpers/run-builds.ps1]: Thanks!

File: azure-pipelines/run-builds-job.yml (hunk 4 above)
Comment (vmapetr) [reply azure-pipelines/run-builds-job.yml]: Or we can simply pass System.AccessToken from definition?

Comment (MaksimZhukov) [reply azure-pipelines/run-builds-job.yml]: Thanks!

File: azure-pipelines/run-builds-job.yml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  - task: PowerShell@2
+    displayName: 'Run build'
+    inputs:
+      targetType: filePath
+      filePath: './helpers/run-builds.ps1 '
+      arguments: '-TeamFoundationCollectionUri $(System.TeamFoundationCollectionUri) `
+                  -AzDoProjectName $(System.TeamProject) `
+                  -AzDoAccessToken $(System.AccessToken) `
+                  -SourceBranch $Env:BUILD_SOURCEBRANCH `
+                  -ToolVersions $Env:PYTHON_VERSIONS `
```
Comment (maxim-lobanov) [inline azure-pipelines/run-builds-job.yml]: I suggest to pass `-ToolVersions "$(PYTHON_VERSIONS)"`. Quotes will handle case with spaces `2.7.0, 3.6.1`

File: azure-pipelines/run-builds-job.yml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+  - task: PowerShell@2
+    displayName: 'Run build'
+    inputs:
+      targetType: filePath
+      filePath: './helpers/run-builds.ps1 '
+      arguments: '-TeamFoundationCollectionUri $(System.TeamFoundationCollectionUri) `
+                  -AzDoProjectName $(System.TeamProject) `
+                  -AzDoAccessToken $(System.AccessToken) `
+                  -SourceBranch $Env:BUILD_SOURCEBRANCH `
```
Comment (maxim-lobanov) [inline azure-pipelines/run-builds-job.yml]: $(Build.SourceBranch)

File: azure-pipelines/run-builds-job.yml [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+                  -ToolVersions $Env:PYTHON_VERSIONS `
+                  -DefinitionId $Env:DEFINITION_ID'
```
Comment (maxim-lobanov) [inline azure-pipelines/run-builds-job.yml]: $(DEFINITION_ID)

File: azure-pipelines/run-builds-job.yml (hunk 6 above)
Comment (MaksimZhukov) [reply azure-pipelines/run-builds-job.yml]: Thanks for the point!

**Pull request merged**